File: c/03-mad-lib/01.c
# include <stdio.h>
# include <string.h>

void getInput(char* variable, int size) {
  fgets(variable, size, stdin);
  variable[strcspn(variable, "\n")] = 0;
}

int main() {
  char noun[50];
  char verb[50];
  char adjective[50];
  char adverb[50];

  printf("Enter a noun: ");
  getInput(noun, 50);

  printf("Enter a verb: ");
  getInput(verb, 50);

  printf("Enter a adjective: ");
  getInput(adjective, 50);

  printf("Enter a adverb: ");
  getInput(adverb, 50);

  printf("Do you %s your %s %s %s? That's hilarious!", verb, adjective, noun, adverb);

  return 0;
}

File: js/04-mad-lib/01.mjs
import * as readline from 'node:readline/promises';
import { stdin, stdout } from 'node:process';

const rl = readline.createInterface({
  input: stdin,
  output: stdout,
})

const noun = await rl.question('Enter a noun: ');
const verb = await rl.question('Enter a verb: ');
const adjective = await rl.question('Enter an adjective: ');
const adverb = await rl.question('Enter an adverb: ');
rl.close();

console.log(`Do you ${verb} your ${adjective} ${noun} ${adverb}? That's hilarous!`);
